### PR TITLE
[cmake][LoongArch] Add testsuite framework for SX/ASX

### PR DESCRIPTION
**Pull request by leecheechen**

### CODE REVIEW

**Comment by leecheechen** [comment]
cc @SixWeining @yjijd

File: cmake/modules/DetectLoongArchSupportVector.cmake [hunk 1]
```diff
@@ -0,0 +1,47 @@
+##===- DetectLoongArchSupportVector.cmake ---------------------------------===##
+#
+# Performs a try_run to determine if the target machine can run the vector
```
Comment (SixWeining) [inline cmake/modules/DetectLoongArchSupportVector.cmake:3]:
```suggestion
# Performs a try_run with a simple program calling LSX/LASX builtin to determine if the LoongArch CPU supports vector instructions.
```

File: cmake/modules/DetectLoongArchFeatureASX.c [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#include <lasxintrin.h>
+
+int main(void) {
+  __m256i a, b, c;
+  c = __lasx_xvand_v(a, b);
+  return 0;
+}
```
Comment (SixWeining) [inline cmake/modules/DetectLoongArchFeatureASX.c:7]: How about renaming this file to `SimpleProgramCallLSXBuiltin.c` which is more semantically consistent?

Comment (SixWeining) [reply cmake/modules/DetectLoongArchFeatureASX.c:7]: Another alternative is writing this temporary file when running cmake. We can refer the `try_run` usage in llvm-project: `llvm-project/llvm/cmake/modules/FindZ3.cmake`.

The pros is we can keep the `cmake/modules` directory as clean as possible.

**Comment by leecheechen** [comment]
Address @SixWeining's comments and thanks for your feedback.

**Review by SixWeining — APPROVED** [review]

**Pull request merged**